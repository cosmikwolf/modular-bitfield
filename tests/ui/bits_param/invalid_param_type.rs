use modular_bitfield::prelude::*;

const NOT_LITERAL: u32 = 32;

#[bitfield(bits = NOT_LITERAL)]
pub struct SignInteger {
    sign: bool,
    value: B31,
}

fn main() {}
