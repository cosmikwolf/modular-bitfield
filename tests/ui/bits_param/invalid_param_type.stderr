error: encountered invalid value argument for #[bitfield] `bits` parameter
 --> tests/ui/bits_param/invalid_param_type.rs:5:19
  |
5 | #[bitfield(bits = NOT_LITERAL)]
  |                   ^^^^^^^^^^^
